tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1144,6 +1244,55 @@ void EnqueueProgramImpl(CommandQueue& cq, std::variant < std::reference_wrapper<
     }, program);
 }
 
+void EnqueueQueueRecordEvent(CommandQueue& cq, Event &event) {
+    TT_ASSERT(cq.id() == 0, "EnqueueQueueRecordEvent only supported on first command queue on device for time being.");
+    TT_ASSERT(event.device == nullptr, "EnqueueQueueRecordEvent expected to be given an uninitialized event");
+    TT_ASSERT(event.event_id == -1, "EnqueueQueueRecordEvent expected to be given an uninitialized event");
+    TT_ASSERT(event.cq_id == -1, "EnqueueQueueRecordEvent expected to be given an uninitialized event");
+
+    detail::DispatchStateCheck(true);
+    cq.run_command(CommandInterface{
+        .type = EnqueueCommandType::ENQUEUE_RECORD_EVENT,
+        .blocking = false,
+        .event = event,
+    });
+}
+
+void EnqueueRecordEventImpl(CommandQueue& cq, std::reference_wrapper<Event> event) {
+    cq.hw_command_queue().enqueue_record_event(event);
+}
+
+
+void EnqueueQueueWaitForEvent(CommandQueue& cq, Event &event) {
+    TT_ASSERT(cq.id() == 0, "EnqueueQueueRecordEvent only supported on first command queue on device for time being.");
```

Comment:
Same, is this restriction necessary? Why can't we support recording events on CQ1?

Reply:
Not necessary, but keeping until I can test with multi-CQ (WIP today, hope to include with this PR).